src/hypertable.c

```diff
@@ -483,6 +483,7 @@ hypertable_insert_relation(Relation rel,
 		values[Anum_hypertable_associated_table_prefix - 1] = NameGetDatum(associated_table_prefix);
 	else
 	{
+		memset(NameStr(default_associated_table_prefix), '\0', NAMEDATALEN);
```

Comment:
FWIW, I don't think this is a real safety issue since `snprintf` will always terminate with a `'\0'`. Looks like something needed to simply quench Valgrind warnings. Maybe this should be reflected in the commit message?

Reply:
I'm not actually positive about this. Various portions of the PG code seem to create Name objects with '\0' padding using palloc0 or whatnot. Maybe this is just their way of preventing valgrind warnings on their end but I wouldn't be too surprised if parts of the code depended on it.

Reply:
Anyway changed message see if you agree.

Reply:
I think the `palloc0` is good practice, e.g., for heap allocated structs when you are not explicitly initializing all fields. For other use cases I am pretty sure this is to quench Valgrind errors. There is a cost to initializing memory, so that is why it is normally optional.